### PR TITLE
[refactor-prompt] Migrate command: wallet split

### PR DESCRIPTION
**Pull request by LysanderGG**
**What current issue(s) does this address, or what feature is it adding?**
#623 
Migrate the command `wallet split` to the new `Command` system.

**How did you solve this problem?**
Implementation of `CommandWalletUnspent`

**How did you make sure your solution works?**
manual testing
Add unit tests 

**Are there any special changes in the code that we should be aware of?**
No

**Please check the following, if applicable:**

- [x] Did you add any tests?
- [x] Did you run `make lint`?
- [x] Did you run `make test`?
- [x] Are you making a PR to a feature branch or development rather than master?
- [ ] Did you add an entry to `CHANGELOG.rst`? (if not, please do)

### CODE REVIEW

**Comment by LysanderGG** [comment]
@ixje This PR is now ready to review.

File: neo/Prompt/Commands/Wallet.py [hunk 1]
```diff
@@ -305,6 +306,66 @@ def command_desc(self):
...
+
+        if len(arguments) > 5:
+            # the 5th argument is the optional attributes,
+            print("Too many parameters supplied. Please check your command")
+            return None
+
+        try:
+            from_addr = wallet.ToScriptHash(arguments[0])
+        except ValueError as e:
+            print(str(e))
```
Comment (ixje) [inline neo/Prompt/Commands/Wallet.py]: I think we might want to add a check with `isValidPublicAddress()` and print a user friendly message. Right now an invalid address gives a confusing error:
```
neo> wallet split bad_addr NEO n 2
subsection not found
```

File: neo/Prompt/Commands/Wallet.py [hunk 2]
```diff
@@ -305,6 +306,66 @@ def command_desc(self):
...
+            return None
+
+        asset_id = PromptUtils.get_asset_id(wallet, arguments[1])
```
Comment (ixje) [inline neo/Prompt/Commands/Wallet.py]: I think we should test for `asset_id` being `None` here and give a user friendly error message that the asset is bad. Right now it gives the following:
```
neo> wallet split AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y bla 0 2
No unspent assets matching the arguments.
```
This initially had me thinking that it was failing on a wrong `unspent_index`.

File: neo/Prompt/Commands/Wallet.py [hunk 3]
```diff
@@ -305,6 +306,66 @@ def command_desc(self):
...
+
+        try:
+            index = int(arguments[2])
```
Comment (ixje) [inline neo/Prompt/Commands/Wallet.py]: Can we also test for a negative value please. The other invalid handling is fine
```
neo> wallet split AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y NEO -1 2
No unspent assets matching the arguments.
neo> wallet split AK2nJJpJr6o664CWJKi1QRXjqeic2zRp8y NEO bla 2
Invalid unspent index value: bla
```

File: neo/Prompt/Commands/Wallet.py [hunk 4]
```diff
@@ -305,6 +306,66 @@ def command_desc(self):
...
+
+        try:
+            divisions = int(arguments[3])
+        except ValueError:
+            print(f"Invalid divisions value: {arguments[3]}")
+            return None
+
+        if divisions < 1:
+            print("Divisions cannot be lower than 1")
+            return None
```
Comment (ixje) [inline neo/Prompt/Commands/Wallet.py]: I think the divisions should not be lower than 2. When I specified `1` it would not split anything. Specifying `2` did work.

File: neo/Prompt/Commands/tests/test_wallet_commands.py [hunk 5]
```diff
@@ -875,3 +874,113 @@ def test_wallet_import_multisig_address(self):
...
+        # test wallet split with not enough arguments
+        with patch('sys.stdout', new=StringIO()) as mock_print:
+            args = ['split', self.wallet_1_addr]
+            res = CommandWallet().execute(args)
+            self.assertIsNone(res)
+            self.assertIn("specify the required parameters", mock_print.getvalue())
+
+        # test wallet split with too much arguments
+        with patch('sys.stdout', new=StringIO()) as mock_print:
+            args = ['split', self.wallet_1_addr, 'neo', 0, 2, 'too', 'much']
```
Comment (ixje) [inline neo/Prompt/Commands/tests/test_wallet_commands.py]: We should probably supply all arguments as strings, as that's what the actual code in `execute()` gets to deal with.

File: neo/Prompt/Commands/Wallet.py [hunk 6]
```diff
@@ -305,6 +306,66 @@ def command_desc(self):
...
+
+        if len(arguments) == 5:
+            fee = Fixed8.TryParse(arguments[4])
```
[the remark this answers is not in the document]
Comment (LysanderGG) [reply neo/Prompt/Commands/Wallet.py]: I didn't know about `require_positive` for `TryParse`, thank you!

**Comment by LysanderGG** [comment]
@ixje @jseagrave21 Thank you both for the quick review.
I've reflected your reviews but one, waiting for confirmation.

File: neo/Prompt/Commands/Wallet.py (hunk 3 above)
Comment (ixje) [reply neo/Prompt/Commands/Wallet.py]: Ok that's fine. Thanks for the explanation

**Comment by ixje** [comment]
🥇
